src/main/java/org/mifos/connector/phee/zeebe/workers/implementation/BatchDetailWorker.java

```diff
@@ -65,4 +100,118 @@ public void setup() {
         });
 
     }
+
+    public BatchDetailResponse callApi(String batchId, int pageNo, int pageSize, String tenant) {
+        // Set up the RestTemplate
+        RestTemplate restTemplate = new RestTemplate();
+
+        // Set headers
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Platform-TenantId", tenant);
+
+        // Construct URL with query parameters
+        String apiUrl = mockPaymentSchemaConfig.batchDetailUrl + "?"
+                + BATCH_ID + "=" + batchId + "&"
+                + PAGE_NO + "=" + pageNo + "&"
+                + PAGE_SIZE + "=" + pageSize;
+
+        // Construct request entity with headers
+        HttpEntity<String> requestEntity = new HttpEntity<>(headers);
+        BatchDetailResponse batchDetailResponse = null;
+
+        try {
+            // Make the API call
+            ResponseEntity<String> response = restTemplate.exchange(
+                    apiUrl,
+                    HttpMethod.GET,
+                    requestEntity,
+                    String.class
+            );
+            batchDetailResponse = objectMapper.readValue(response.getBody(), BatchDetailResponse.class);
+
+            // Log response
+            logger.info("Batch detail API response: \n\n" + response.getBody());
```

Comment:
1. Use debug-level logs to print payload
2. Use {} instead of + for concatenation